### PR TITLE
fix/Add token from context in API calls

### LINKED ISSUE #238: Add token from context in API calls

### PR DESCRIPTION
**Pull request by Navneet-Sahota**
close #238

## Types of changes

- [x] Add tags to auto close the corresponding issue.
- [x] I followed branch and commit naming convention.
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: client/src/API/index.js [hunk 1]
```diff
@@ -35,9 +35,12 @@ const updatePost = async (body, postId) => {
   }
 };
 
-const getCategories = async () => {
+const getCategories = async (token = undefined) => {
   try {
     getToken();
```
Comment (avinash360) [inline client/src/API/index.js]: put the getToken function in else why should we set token twice

File: client/src/API/index.js [hunk 2]
```diff
@@ -84,9 +87,12 @@ const signin = async (body) => {
   }
 };
 
-const getTop10Posts = async () => {
+const getTop10Posts = async (token = undefined) => {
   try {
     getToken();
```
Comment (avinash360) [inline client/src/API/index.js]: same here.

File: client/src/API/index.js [hunk 3]
```diff
@@ -35,9 +35,12 @@ const updatePost = async (body, postId) => {
   }
 };
 
-const getCategories = async () => {
+const getCategories = async (token = undefined) => {
```
Comment (avinash360) [inline client/src/API/index.js]: if token is not provided then its by default undefined

File: client/src/API/index.js [hunk 4]
```diff
@@ -84,9 +87,12 @@ const signin = async (body) => {
   }
 };
 
-const getTop10Posts = async () => {
+const getTop10Posts = async (token = undefined) => {
```
Comment (avinash360) [inline client/src/API/index.js]: same here

File: client/src/Routes.jsx [hunk 5]
```diff
@@ -72,16 +72,16 @@ function Routes() {
   const { state, dispatch } = useContext(GlobalContext);
 
   useEffect(() => {
-    try {
-      if (state.user && Object.keys(state.user).length) {
+    if (state.user && Object.keys(state.user).length) {
```
Comment (avinash360) [inline client/src/Routes.jsx]: put this if statement in try catch, if it throws error you wont be able to handle it

File: client/src/API/index.js (hunk 3 above)
Comment (Navneet-Sahota) [reply client/src/API/index.js]: Whats wrong with this?

Comment (avinash360) [reply client/src/API/index.js]: nothing wrong, its just that it is obvious that token will be undefined if it is not provided

**Pull request merged**